underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -208,17 +207,49 @@ public static S3AUnderFileSystem createInstance(AlluxioURI uri,
       clientConf.setSignerOverride(conf.get(PropertyKey.UNDERFS_S3_SIGNER_ALGORITHM));
     }
 
-    AmazonS3ClientBuilder clientBuilder = AmazonS3Client.builder()
-        .withCredentials(credentials)
+    AwsClientBuilder.EndpointConfiguration endpointConfiguration
+        = createEndpointConfiguration(conf, clientConf);
+
+    AmazonS3 amazonS3Client
+        = createAmazonS3(credentials, clientConf, endpointConfiguration, conf);
+
+    ExecutorService service = ExecutorServiceFactories
+        .fixedThreadPool("alluxio-s3-transfer-manager-worker",
+            numTransferThreads).create();
+
+    TransferManager transferManager = TransferManagerBuilder.standard()
+        .withS3Client(createAmazonS3(credentials, clientConf, endpointConfiguration, conf))
+        .withExecutorFactory(() -> service)
+        .withMultipartCopyThreshold(MULTIPART_COPY_THRESHOLD)
+        .build();
+
+    return new S3AUnderFileSystem(uri, amazonS3Client, bucketName,
+        service, transferManager, conf, streamingUploadEnabled);
+  }
+
+  /**
+   * Create an AmazonS3 client.
+   *
+   * @param credentialsProvider the credential provider
+   * @param clientConf the client config
+   * @param endpointConfiguration the endpoint config
+   * @param conf the Ufs config
+   * @return the AmazonS3 client
+   */
+  public static AmazonS3 createAmazonS3(AWSCredentialsProvider credentialsProvider,
+      ClientConfiguration clientConf,
+      AwsClientBuilder.EndpointConfiguration endpointConfiguration,
+      UnderFileSystemConfiguration conf) {
+    AmazonS3ClientBuilder clientBuilder = AmazonS3ClientBuilder
+        .standard()
```

Comment:
why add `.standard()`?
This line is not tested by me or @yuzhu so we are not quite sure whether it will break something or not.
S3 client with the S3 region/endpoint is kind of tricky.

Reply:
It is my mistaken. It should be 100% same with before.